app/src/main/java/org/fossasia/susi/ai/signup/SignUpActivity.kt

```diff
@@ -40,13 +41,20 @@ class SignUpActivity : AppCompatActivity(), ISignUpView {
             email.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[0].toString())
             password.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[1].toString())
             confirm_password.editText?.setText(savedInstanceState.getCharSequenceArray(Constant.SAVED_STATES)[2].toString())
-            if (savedInstanceState.getBoolean(Constant.SERVER)) {
-                input_url.visibility = View.VISIBLE
+
+          if(savedInstanceState.getBoolean(Constant.SERVER)) {
+                input_url_sign_up.visibility = View.VISIBLE
             } else {
-                input_url.visibility = View.GONE
+                input_url_sign_up.visibility = View.GONE
             }
         }
 
+        var bundle: Bundle? = intent.extras;
+        if (bundle != null) {
+            var string = bundle.getString("email")
+            email.editText?.setText(string)
```

Comment:
Same here

Reply:
@iamareebjamal where exactly the space? bcoz there is space after if here

Reply:
> There is no guarantee that email will be in this bundle, this should be nullable